src/lib/dataviews/histogram/Histogram.ts

```diff
@@ -13,27 +14,34 @@ export class HistogramDataView extends DataViewWrapper {
     column: string,
     options: HistogramDataViewOptions
   ) {
-    let dataView;
     const { mode } = options;
 
-    switch (mode) {
-      case DataViewCalculation.LOCAL: {
-        dataView = new DataViewLocal(dataSource as Layer, column);
-        break;
-      }
+    const dataViewCreationFn =
+      createDataView[mode as DataViewCalculation] ||
+      createDataView[DataViewCalculation.REMOTE_FILTERED];
+    const dataView = dataViewCreationFn(dataSource, column);
 
-      case DataViewCalculation.REMOTE: {
-        dataView = new DataViewRemote(dataSource as Source, column);
-        break;
-      }
+    this.dataviewImpl = new HistogramImpl(dataView, options);
+  }
+}
+
+const createDataView = {
+  [DataViewCalculation.LOCAL](dataSource: Layer | Source, column: string) {
+    return new DataViewLocal(dataSource as Layer, column);
+  },
 
-      default: {
-        dataView = new DataViewRemote(dataSource as Layer, column);
-        dataView.addFilter(`VIEWPORT_FILTER_${uuidv4()}`, BuiltInFilters.VIEWPORT);
-        break;
-      }
+  [DataViewCalculation.REMOTE](dataSource: Layer | Source, column: string) {
+    if (isGeoJSONSource(dataSource)) {
+      const useViewport = false;
+      return new DataViewLocal(dataSource as Layer, column, useViewport);
     }
 
-    this.dataviewImpl = new HistogramImpl(dataView, options);
+    return new DataViewRemote(dataSource as Source, column);
+  },
+
+  [DataViewCalculation.REMOTE_FILTERED](dataSource: Layer | Source, column: string) {
+    const dataView = new DataViewRemote(dataSource as Layer, column);
```

Comment:
Don't you need to avoid adding the filter in geoJSON case?

Reply:
Yes, you are right, I missed that second if in the original code. We need to avoid adding the filter and creating a remote DataView which makes no sense.

I was wondering it it would be better to throw an error when using a GeoJSON source with any unsupported mode, like `REMOTE_FILTERED` mode rather than "fixing it" automagically.